### ISSUE TITLE
CVODE solver gives wrong results when stepSize is large when simulating FMU's in CoSimulation Mode

### ISSUE THREAD

**Issue by arun3688** [OpenModelica/OpenModelica#11549]
### Related issues

https://github.com/OpenModelica/OMSimulator/issues/1269

### Description

CVODE solver gives wrong results when large stepSizes are chosen. 

### Steps to Reproduce
Consider the following model 

```
loadString("
model stiffProblem
  Real x(start=x0, fixed=true);
  Real solution;
  parameter Real x0 = 1;
equation
  der(x)=-4*(x-2);
  solution = 2-exp(-4*time);
end stiffProblem;
"); getErrorString();
setCommandLineOptions("-d=newInst --fmiFlags=s:cvode"); getErrorString();
buildModelFMU(stiffProblem, version = "2.0", fmuType="me_cs"); getErrorString();
```
or from the testsuite path the mos script is located at `OpenModelica/testsuite/openmodelica/fmi/CoSimulation/2.0/simpleStiffFMU.mos`

and then run the fmu using OMSimulator with the following settings

`>>> OMSimulator stiffProblem.fmu --stopTime=250 --stepSize=0.5 --mode=cs  --resultFile="stiffProblem_Dynamic_res.mat" --progressBar=true
`

### Wrong Result
```
val(x, 250.0, "stiffProblem_Dynamic_res.mat"); getErrorString();
// Result
// 0.9999999999722797
```
### Expected Results

```
val(x, 250.0, "stiffProblem_Dynamic_res.mat"); getErrorString();
// Result
// 2.0 
```

### Version and OS

 - OpenModelica Version: v1.23.0-dev-117-gf075b4ea32-cmake
 - OS: [Windows 10, 64 bit]

**Comment by casella** [comment]
@AnHeuermann I need you help here, please. In the [FMI export user's guide](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/fmitlm.html#fmi-export), I read

>  For the Co-Simulation FMU two integrator methods are available:
>
 >   Forward Euler [default]
>
 >   SUNDIALS CVODE (see [[1]](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/fmitlm.html#f1))
>
>Forward Euler uses root finding, which does an Euler step of communicationStepSize in fmi2DoStep. Events are checked for before and after the call to fmi2GetDerivatives.
>
> If CVODE is chosen as integrator the FMU should also include runtime dependencies ([--fmuRuntimeDepends=modelica](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/omchelptext.html#omcflag-fmuruntimedepends)) to copy all used dynamic libraries into the generated FMU to make it exchangeable.

The description of what the Forward Euler does is pretty complete. On the other hand, what exactly happens when CVODE is selected is not at all clear. The reference to the Sundials webpage is broken and we should probably update it. In the CVODE intro page https://computing.llnl.gov/projects/sundials/cvode I read:

>  For nonstiff problems, CVODE includes the Adams-Moulton formulas, with the order varying between 1 and 12. For stiff problems, CVODE includes the Backward Differentiation Formulas (BDFs) in so-called fixed-leading coefficient form, with order varying between 1 and 5.

There are several things that are not clear to me, when I set `--fmiFlags=s:cvode`:
- is the non-stiff Adams-Moulton or the stiff BDF solver used
- which order is selected for the algorithm? Is it fixed or variable? Can it be set?
- when fmi2DoStep is called, is just one step of the integration algorithm (whatever it is) executed, with a step lenght equal to the Interval parameter, or is a variable time step simulation started  with some error tolerance, running until the end of the Interval time span?
- in the latter case, how is the error tolerance set?

I think we should make this clear in the user's guide - one should not need to look at the source code to figure out. 

Probably the default selection which is currently taken when setting `--fmiFlags=s:cvode` picks the Adams-Moulton formula and tries to do one step with length equal to Interval. The eigenvalue of the test problem posted by @arun3688 is -4, the product lambda*h with a time step of 0.5 is -2, which is not numerically stable for order >= 4.

@AnHeuermann can you please shed some light?

**Comment by AnHeuermann** [comment]
Yeah, I should update the User's Guide. I summarized some defaults in https://github.com/OpenModelica/OpenModelica/issues/11662#issuecomment-1835794958.

Basically the default values of the `-cvode*` [runtime flags](https://openmodelica.org/doc/OpenModelicaUsersGuide/latest/simulationflags.html) are used and they can't be changed.

**Comment by casella** [comment]
OK, thanks, I had forgotten about #11662. I think we should make it a bit flexible, it is a shame that we are bound to the default values of CVODE only.

**Comment by casella** [comment]
For sure we should probably use a default which is robust enough to handle stiff systems. If one doesn't use Euler for co-simulation, it is usually because of that.
